src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/GetPackageReferenceUtility.cs

```diff
@@ -167,5 +174,58 @@ private static IReadOnlyList<PackageDependency> GetTransitivePackagesForLibrary(
 
             return null;
         }
+
+        internal static TransitivePackageReference MergeTransitiveOrigin(PackageReference currentPackage, TransitiveEntry transitiveEntry)
+        {
+            if (currentPackage == null)
+            {
+                throw new ArgumentNullException(nameof(currentPackage));
+            }
+            if (transitiveEntry == null)
+            {
+                throw new ArgumentNullException(nameof(transitiveEntry));
+            }
+
+            var transitiveOrigins = new SortedSet<PackageReference>(PackageReferenceMergeComparer);
+            transitiveEntry?.Keys?.ForEach(fwRuntimePair =>
+            {
+                if (fwRuntimePair != null)
+                {
+                    IList<PackageReference> fwTransitiveOrigins = transitiveEntry[fwRuntimePair];
+                    if (fwTransitiveOrigins != null)
+                    {
+                        foreach (PackageReference transitiveOrigin in fwTransitiveOrigins)
+                        {
+                            if (transitiveOrigin != null && transitiveOrigin.PackageIdentity != null && transitiveOrigin.PackageIdentity.Id != null && transitiveOrigin.PackageIdentity.Version != null)
+                            {
+                                transitiveOrigins.Add(transitiveOrigin);
+                            }
+                        }
+                    }
+                }
+            });
+
+            List<PackageReference> merged;
+            if (transitiveOrigins.Any())
+            {
+                merged = transitiveOrigins
+                // unique packageId per project
+                .GroupBy(tr => tr.PackageIdentity.Id)
+                // highest version found
+                .Select(g => g.OrderByDescending(pr => pr.PackageIdentity.Version).First())
+                .ToList();
+            }
+            else
+            {
+                merged = new();
```

Comment:
Do we have to create instance of list?
Could simply return null then check it before use next time?

Reply:
Use Enumerable.Empty instead of allocating a new list.